src/Text/Pandoc/Writers/HTML.hs

```diff
@@ -475,6 +475,9 @@ obfuscateLink opts attr (renderHtml -> txt) s =
                     then ("e", name' ++ " at " ++ domain')
                     else ("'" ++ obfuscateString txt ++ "'",
                           txt ++ " (" ++ name' ++ " at " ++ domain' ++ ")")
+              (_, classNames, _) = attr
+              classNamesStr = (if not $ null classNames then " " else "")
+                           ++ intercalate " " classNames
```

Comment:
Perhaps more economical?
```
classNamesStr = concatMap (' ':) classNames
```
I think that will do the same, without intercalate and the conditional.

Also, a stylistic thing, which may just be a quirk of my own:  I find the `$` in

    if not $ null classnames then " " else ""

to be a bit ugly -- usually I only use the `$` when it is clear to the eye where its scope ends (parentheses or end ofline).  Here the scope ends before `then`, but there's no visual marker for that.  So I'd tend to use `(null classNames)` instead.  Of course what you have is perfectly correct, this is just a stylistic thing.